test/e2e/sync/sync.go

```diff
@@ -0,0 +1,449 @@
+	}
+
+	for i := 0; i < NoOfObjects; i++ {
+		content = generateRandomString(100)
+		fileName = FolderName + "/" + strings.ToLower(generateRandomString(6)) + ".txt"
+		f, err := os.Create(fileName)
+
+		if err != nil {
+			fmt.Println("ERROR: ", err)
+			return false
+		}
+		defer f.Close()
+
+		_, err = f.WriteString(content)
+		if err != nil {
+			fmt.Println("ERROR: ", err)
+			return false
+		}
+	}
+	return true
+}
+
+func deleteFiles() bool {
+	fmt.Println("STEP: Delete created Files")
+	err := os.RemoveAll(FolderName)
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+	err = os.RemoveAll("spec.yaml")
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+	return true
+}
+
+func uploadObjects(spec Spec) bool {
+
+	fmt.Println("STEP: UploaOjects to Buckets")
+	var filePath string
+	files, err := ioutil.ReadDir(FolderName)
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+
+	bxCli, err := client.NewClientWithEnv(APIKey, client.DefaultEnv, Debug)
+
+	if err != nil {
+		fmt.Println("ERROR: ", err)
+		return false
+	}
+
+	for _, src := range spec {
+
+		s3Cli, err := client.NewS3Client(bxCli, src.Cos, src.Region)
+		if err != nil {
+			fmt.Println("ERROR: ", err)
+			return false
+		}
+		for _, f := range files {
+			filePath = FolderName + "/" + f.Name()
+			fmt.Println("STEP: Uploading File ", filePath, " To Bucket ", src.Bucket)
+			err = s3Cli.UploadObject(filePath, f.Name(), src.Bucket)
+			if err != nil {
+				fmt.Println("ERROR: ", err)
+				return false
+			}
+		}
+	}
+
+	return true
+
+}
+
+func createResources(spec Spec) bool {
+
+	if !createSpecFile(spec) {
+		return false
+	}
+
+	if !createCOSInstances(spec) {
+		return false
+	}
+
+	if !createBuckets(spec) {
+		return false
+	}
+
+	if !createFiles() {
+		return false
+	}
+
+	if !uploadObjects(spec) {
+		return false
+	}
+
+	return true
+}
+
+func deleteResources(spec Spec) bool {
+	if !deleteCOSInstances(spec) {
+		return false
+	}
+
+	if !deleteFiles() {
+		return false
+	}
+	return true
+}
+
+var _ = CMDDescribe("pvsadm image sync tests", func() {
+
+	It("run with --help option", func() {
+		status, stdout, stderr := runSyncCMD(
+			"--help",
+		)
+		Expect(status).To(Equal(0))
+		Expect(stderr).To(Equal(""))
+		Expect(stdout).To(ContainSubstring("Examples:"))
+	})
+
+	framework.NegativeIt("run without spec-file flag", func() {
+		status, _, stderr := runSyncCMD()
+		Expect(status).NotTo(Equal(0))
+		Expect(stderr).To(ContainSubstring(`"spec-file" not set`))
+	})
+
+	framework.NegativeIt("run with yaml file that doesn't exist", func() {
+		status, _, stderr := runSyncCMD("--spec-file", "fakefile.yaml")
+		Expect(status).NotTo(Equal(0))
+		Expect(stderr).To(ContainSubstring(`no such file or directory`))
+	})
+
+	It("Copy Object Between Buckets", func() {
+		spec := generateStruct()
+		res := createResources(spec)
+		Expect(res).To(BeTrue())
+		status, _, _ := runSyncCMD("--spec-file", "spec.yaml")
+		Expect(status).To(Equal(0))
```

Comment:
We may need to add a check to see the objects really got copied to the target bucket or not?!

Reply:
Verification method added to verify the objects copied to Buckets